cherab/solps/solps_plasma.py

```diff
@@ -32,139 +31,219 @@
+        self._inside_mesh = inside_outside
 
-        self.mesh = mesh
+        if not len(species_list):
+            raise ValueError('Argument "species_list" must contain at least one species.')
+        self._species_list = tuple(species_list)  # adding additional species is not allowed
 
         self._electron_temperature = None
         self._electron_density = None
-        self._species_list = None
+        self._ion_temperature = None
+        self._neutral_temperature = None
         self._species_density = None
-        self._rad_par_flux = None
-        self._radial_area = None
-        self._b2_neutral_densities = None
-        self._velocities_parallel = None
-        self._velocities_radial = None
-        self._velocities_toroidal = None
+        self._velocities = None
         self._velocities_cartesian = None
-        self._inside_mesh = None
         self._total_rad = None
         self._b_field_vectors = None
         self._b_field_vectors_cartesian = None
-        self._parallel_velocities = None
-        self._radial_velocities = None
         self._eirene_model = None
         self._b2_model = None
         self._eirene = None
 
     @property
-    def electron_temperature(self):
+    def mesh(self):
         """
-        Simulated electron temperatures at each mesh cell.
+        SOLPSMesh instance.
         :return:
         """
-        return self._electron_temperature
+        return self._mesh
 
     @property
-    def electron_density(self):
+    def species_list(self):
         """
-        Simulated electron densities at each mesh cell.
+        Tuple of species elements in the form (species name, charge).
         :return:
         """
-        return self._electron_density
+        return self._species_list
 
     @property
-    def species_list(self):
+    def electron_temperature(self):
         """
-        Text list of species elements present in the simulation.
+        Simulated electron temperatures at each mesh cell.
         :return:
         """
-        return self._species_list
+        return self._electron_temperature
+
+    def get_electron_temperature(self):
+
+        return self._electron_temperature
+
+    def set_electron_temperature(self, value):
+        _check_array("electron_temperature", value, (self.mesh.nx, self.mesh.ny))
+
+        self._electron_temperature = value
 
     @property
-    def species_density(self):
+    def ion_temperature(self):
         """
-        Array of species densities at each mesh cell.
+        Simulated ion temperatures at each mesh cell.
         :return:
         """
-        return self._species_density
+        return self._ion_temperature
+
+    def get_ion_temperature(self):
+
+        return self._ion_temperature
+
+    def set_ion_temperature(self, value):
+        _check_array("ion_temperature", value, (self.mesh.nx, self.mesh.ny))
+
+        self._ion_temperature = value
 
     @property
-    def radial_particle_flux(self):
```

Comment:
I don't understand why the @property access was removed. Setter and getters are not python style. These really should be properties:

```
@property
def myvar(self):
    ......

@myvar.setter(self, value):
    ......
```